src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
```

Comment:
Ah, I see. Like a general substitution framework? Yeah, the goal here was just a one-off custom substitution in which it *only* knows about a couple properties. But definitely possibilities. The only thing then would be choosing awkward names like `%{group}` and `%{name}` (instead of `%{trace}`) so that it's backward compatibility in case we substitute trace properties instead of special/custom/ad hoc keys.